Makefile

```diff
@@ -2,13 +2,19 @@ all: check build test
 
 export RUSTFLAGS=-Dwarnings -Dclippy::all -Dclippy::pedantic
 
-install:
+# update the Cargo.lock every time the Cargo.toml changes.
+Cargo.lock: Cargo.toml
+	cargo update
```

Comment:
❓ Why do we need to run `cargo update` everytime the toml file changes? That will result in updating all transitive dependencies, which is unlikely the intent of someone making a change to Cargo.toml. If someone wants to update the transitive dependencies they can run `cargo update` manually.

For context on why I'm interested in this: We usually try to be very much in control and intentional with updating dependencies, because from a security point-of-view doing so can be a risk.

Reply:
I added this when I found that git versioning changes that I've applied to the `Cargo.toml` file were not taking affect until the `Cargo.lock` file was regenerated. I think that every time we update our dependencies, we'll need to do that.
I'm not really sure if this is required or not, and would be perfectly fine dropping this if it causes any issues.

Reply:
Updating dependencies automatically (unintended) seems risky to me as well.